src/server/gc_worker/compaction_filter.rs

```diff
@@ -146,14 +146,21 @@ impl CompactionFilterFactory for WriteCompactionFilterFactory {
 
         let name = CString::new("write_compaction_filter").unwrap();
         let db = Arc::clone(&gc_context.db);
-        let filter = Box::new(WriteCompactionFilter::new(db, safe_point, context));
+        let filter = Box::new(WriteCompactionFilter::new(
```

Comment:
Do we need to check `too_many_delete_marks` to decide to create compaction filter?

Reply:
It's not necessary. If it can pass the checker `too many delete marks`, it must can pass `check_need_gc`.